compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -827,6 +831,49 @@ impl InitMask {
     }
 }
 
+/// Yields [`InitChunk`]s. See [`InitMask::range_as_init_chunks`].
+pub struct InitChunkIter<'a> {
```

Comment:
There seems to be some conceptual overlap between this and `InitMaskCompressed`... but I am not sure if there is a way to actually share code here.

Reply:
I changed the code that creates `InitMaskCompressed`s to use this...it's not much simpler, but hopefully it'll be faster

Reply:
Nice! It looks like `compress_uninit_range` is now essentially a more efficient version of `range_as_init_chunks(...).collect::<Vec<_>>`. Can you document it as such? Also it would then make sense to move it *below* `InitChunkIter` in this file.

`InitChunkIter` also seems spread across multiple parts of the file; is it possible to make that code all adjacent to each other?